nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/main/java/org/apache/nifi/processors/flume/FlumeSourceProcessor.java

```diff
@@ -0,0 +1,203 @@
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+    public static final PropertyDescriptor SOURCE_NAME = new PropertyDescriptor.Builder()
+        .name("Source Name")
+        .description("The name of the source used in the Flume source configuration")
+        .required(true)
+        .defaultValue("src-1")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+    public static final PropertyDescriptor FLUME_CONFIG = new PropertyDescriptor.Builder()
+        .name("Flume Configuration")
+        .description("The Flume configuration for the source copied from the flume.properties file")
+        .required(true)
+        .defaultValue("")
+        .addValidator(Validator.VALID)
+        .build();
+
+    public static final Relationship SUCCESS = new Relationship.Builder().name("success")
+        .build();
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    private volatile Source source;
+
+    private final NifiSessionChannel pollableSourceChannel = new NifiSessionChannel(SUCCESS);
+    private final AtomicReference<ProcessSessionFactory> sessionFactoryRef = new AtomicReference<>(null);
+    private final AtomicReference<EventDrivenSourceRunner> runnerRef = new AtomicReference<>(null);
+    private final AtomicReference<NifiSessionFactoryChannel> eventDrivenSourceChannelRef = new AtomicReference<>(null);
+    private final AtomicReference<Boolean> stopping = new AtomicReference<>(false);
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        this.descriptors = ImmutableList.of(SOURCE_TYPE, AGENT_NAME, SOURCE_NAME, FLUME_CONFIG);
+        this.relationships = ImmutableSet.of(SUCCESS);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @OnScheduled
+    public void onScheduled(final SchedulingContext context) {
+        try {
+            stopping.set(false);
+            source = SOURCE_FACTORY.create(
+                context.getProperty(SOURCE_NAME)
+                .getValue(),
+                context.getProperty(SOURCE_TYPE)
+                .getValue());
+
+            String flumeConfig = context.getProperty(FLUME_CONFIG)
+                .getValue();
+            String agentName = context.getProperty(AGENT_NAME)
+                .getValue();
+            String sourceName = context.getProperty(SOURCE_NAME)
+                .getValue();
+            Configurables.configure(source,
+                getFlumeSourceContext(flumeConfig, agentName, sourceName));
+
+            if (source instanceof PollableSource) {
+                source.setChannelProcessor(new ChannelProcessor(
+                    new NifiChannelSelector(pollableSourceChannel)));
+                source.start();
```

Comment:
I don't think this should be started until onTrigger so the ProcessSessionFactory can be set. Otherwise, the channel could start sending events to this source, which will immediately fail because there is no session.